shell/platform/darwin/ios/framework/Source/accessibility_bridge.mm

```diff
@@ -247,17 +255,27 @@ - (BOOL)accessibilityPerformMagicTap {
     // TODO(abarth): Use extract once we're at C++17.
   }
 
-  ReleaseObjects(doomed_objects);
-  doomed_objects.clear();
-
-  if (root) {
-    if (!view_.accessibilityElements) {
-      view_.accessibilityElements = @[ root ];
+  SemanticsObject* doomed_focused_object = nil;
+  for (const auto& entry : doomed_objects) {
+    SemanticsObject* object = entry.second;
+    if ([object accessibilityElementIsFocused]) {
+      doomed_focused_object = object;
+      break;
     }
+  }
+
+  if (doomed_focused_object != nil) {
+    // Previously focused element is no longer in the tree.
+    // Passing `nil` as argument to let iOS figure out what to focus next.
```

Comment:
From the Apple docs it wasn't clear to my what the difference between `UIAccessibilityScreenChangedNotification` and `UIAccessibilityLayoutChangedNotification` was supposed to be. After some searching, I found [this post](https://stackoverflow.com/questions/27797515/actual-difference-between-uiaccessibilitylayoutchangednotification-and-uiaccessi) which describes it pretty good.